tests/functional/spec/components/checkbox/checkbox_multiple_detail_answers.spec.js

```diff
@@ -0,0 +1,110 @@
+const helpers = require('../../../helpers');
+
+const MandatoryCheckboxPage = require('../../../generated_pages/checkbox_multiple_detail_answers/mandatory-checkbox.page');
+const SummaryPage = require('../../../generated_pages/checkbox_multiple_detail_answers/summary.page');
+
+describe('Checkbox with multiple "detail_answer" options', function() {
+
+  const checkbox_schema = 'test_checkbox_multiple_detail_answers.json';
+
+  it('Given detail answer options are available, when the user clicks an option the detail answer input should be visible.', function() {
+    return helpers.openQuestionnaire(checkbox_schema).then(() => {
+      return browser
+        .click(MandatoryCheckboxPage.yourChoice())
+        .isVisible(MandatoryCheckboxPage.yourChoiceText()).should.eventually.be.true
+        .click(MandatoryCheckboxPage.cheese())
+        .isVisible(MandatoryCheckboxPage.cheeseText()).should.eventually.be.true;
+    });
+  });
+
+  it('Given a mandatory detail answer, When I select the option but leave the input field empty and submit, Then an error should be displayed.', function() {
+    // Given
+    return helpers.openQuestionnaire(checkbox_schema).then(() => {
+      return browser
+      // When
+        .click(MandatoryCheckboxPage.yourChoice())
+        .click(MandatoryCheckboxPage.submit())
+      // Then
+        .isVisible(MandatoryCheckboxPage.error()).should.eventually.be.true
+        .getText(MandatoryCheckboxPage.errorNumber(1)).should.eventually.contain('Enter your topping choice to continue');
+    });
+  });
+
+  it('Given a mandatory checkbox answer, when there is an error on the page for the detail answer field I enter valid value and submit page, then the error is cleared and I navigate to next page.', function() {
```

Comment:
I think this would read more clearly if some content was moved to the given - "Given a selected  checkbox answer with an error for a mandatory detail answer, When I enter valid value and submit the page, Then ..."